engine/apps/alerts/escalation_snapshot/snapshot_classes/escalation_snapshot.py

```diff
@@ -71,6 +79,20 @@ def next_active_escalation_policy_order(self) -> int:
             next_order = self.last_active_escalation_policy_order + 1
         return next_order
 
+    def next_step_eta_is_valid(self) -> typing.Union[None, bool]:
+        """
+        `next_step_eta` should never be less than the current time (with a 5 minute buffer provided)
+        as this field should be updated as the escalation policy is executed over time. If it is, this means that
+        an escalation policy step has been missed, or is substantially delayed
+
+        if `next_step_eta` is `None` then `None` is returned, otherwise a boolean is returned
+        representing the result of the time comparision
+        """
+        if self.next_step_eta is None:
+            # TODO: this is a valid case I assume?
+            return None
+        return self.next_step_eta > (timezone.now() - timezone.timedelta(minutes=5))
```

Comment:
debatable whether or not it makes more sense to have this method belong inside the celery task file, rather than here on `EscalationSnapshot`